bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/context/ObsContext.java

```diff
@@ -273,6 +273,16 @@ public Coordinates getBaseCoordinates() {
         return new Coordinates(raDeg, decDeg);
     }
 
+    public Option<Coordinates> getBaseCoordinatesOpt() {
+        final Option<Long> when = getSchedulingBlock().map(SchedulingBlock::start);
+        SPTarget target = targets.getBase();
+        return
+            target.getTarget().getRaDegrees(when).flatMap(raDeg ->
+            target.getTarget().getDecDegrees(when).map(decDeg ->
+                new Coordinates(raDeg, decDeg)
+            ));
+    }
+
```

Comment:
The non-`Option` version will be removed in a later PR and this one will be renamed simply to `getBaseCoordinates`.